### PR TITLE
[Issue 669] - Model Entity

### PR DESCRIPTION
**Pull request by pmbrull**
Based on #669, this PR adds the changes required for the first version of a Model Entity. Its purpose is to track any relevant information for ML models to help integrate those services more tightly in any Data Platform.

This first approach handles models as Entities with two interesting fields:
- `algorithm`: Denoting the method to build the model (e.g. regression, SVM, etc.)
- `dashboard`: As a Reference Entity pointing us to the resource tracking the model performance evolution metric(s).

> Further work would tackle the introduction of yet another new Entity: `Feature`, to have detailed information on the data used to train the model, its precedence and any techniques applied on top of it (e.g. normalisation).

Added some tests using the `Dashboard` Entity as an example. Moreover, I spun up the service locally and a simple Model example looks as follows:

```
> http POST http://localhost:8585/api/v1/models < localtests/model.json
HTTP/1.1 201 Created
Content-Length: 431
Content-Type: application/json
Date: Tue, 05 Oct 2021 14:40:14 GMT
Location: http://localhost:8585/api/v1/models/27735538-fec7-438e-b498-f8504d53746f

{
    "algorithm": "regression",
    "dashboard": {
        "href": "http://localhost:8585/api/v1/dashboards/d2b0af00-f419-4905-bb43-036697ce53a5",
        "id": "d2b0af00-f419-4905-bb43-036697ce53a5",
        "type": "dashboard"
    },
    "description": "test model",
    "displayName": "my-model",
    "fullyQualifiedName": "my-model-2",
    "href": "http://localhost:8585/api/v1/models/27735538-fec7-438e-b498-f8504d53746f",
    "id": "27735538-fec7-438e-b498-f8504d53746f",
    "name": "my-model-2",
    "tags": []
}
```

### Type of change :

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

### Checklist:
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### Reviewers
Hi @sureshms, would love your input on this. I am also planning to add a couple of extra tests on the interaction between `Model` and `Dashboard` Entity Reference.

Thanks 🙏

### CODE REVIEW

**Comment by harshach** [comment]
@pmbrull added a couple of comments and can you address the check style comments on the ModelResourceTest.
Overall this looks good. Thanks.

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/models/ModelResource.java [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+  }
+
+
+  @POST
+  @Operation(summary = "Create a model", tags = "models",
+          description = "Create a new model.",
+          responses = {
+                  @ApiResponse(responseCode = "200", description = "The model",
+                          content = @Content(mediaType = "application/json",
+                          schema = @Schema(implementation = Model.class))),
```
[the remark this answers is not in the document]
Comment (pmbrull) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/resources/models/ModelResource.java:212]: Nice catch, thanks! I've noticed that there are other `EntityResource` that are missing the `Create` here, such as `Dashboard` and `Topic`. I'll open a separate issue to review them.

Thanks!

**Review by harshach — APPROVED** [review]
Thanks @pmbrull. Looks great. +1.

**Comment by harshach** [comment]
@pmbrull merged the PR. If you have any todo's from this can we please file follow-up issues.

**Pull request merged**